examples/bme-cdr-example.cpp

```diff
@@ -0,0 +1,247 @@
+        modify.setInteger (AccountType, 1);
+        modify.setString (Symbol, "CSCO");
+        modify.setSide (GWC_SIDE_BUY);
+        modify.setQty (50);
+        modify.setOrderType (GWC_ORDER_TYPE_LIMIT);
+
+        cdrDateTime tt;
+        getTime (tt);
+        modify.setDateTime (TransactTime, tt);
+
+        if (!mGwc->sendModify (modify))
+            mLog->info ("failed to send modify ...");
+    }
+
+    virtual void onOrderRejected (uint64_t seqno, const cdr& msg)
+    {
+        mLog->info ("onOrderRejected msg...");
+        mLog->info ("%s", msg.toString ().c_str ());
+    }
+
+    virtual void onOrderDone (uint64_t seqno, const cdr& msg)
+    {
+        mLog->info ("onOrderDone msg...");
+        mLog->info ("%s", msg.toString ().c_str ());
+    }
+
+    virtual void onOrderFill (uint64_t seqno, const cdr& msg)
+    {
+        mLog->info ("onOrderFill msg...");
+        mLog->info ("%s", msg.toString ().c_str ());
+    }
+
+    virtual void onModifyAck (uint64_t seqno, const cdr& msg)
+    {
+        mLog->info ("onModifyAck msg...");
+        mLog->info ("%s", msg.toString ().c_str ());
+
+        string origcloid;
+        msg.getString (ClOrdID, origcloid);
+
+        gwcOrder cancel;
+        cancel.setString (OrigClOrdID, origcloid);
+        cancel.setString (ClOrdID, "C12345678");
+        cancel.setString (Symbol, "CSCO");
+        cancel.setSide (GWC_SIDE_BUY);
+
+        cdrDateTime tt;
+        getTime (tt);
+        cancel.setDateTime (TransactTime, tt);
+
+        if (!mGwc->sendCancel (cancel))
+            mLog->info ("failed to send cancel ...");
+    }
+
+    virtual void onModifyRejected (uint64_t seqno, const cdr& msg)
+    {
+        mLog->info ("onModifyRejected msg...");
+        mLog->info ("%s", msg.toString ().c_str ());
+    }
+
+    virtual void onCancelRejected (uint64_t seqno, const cdr& msg)
+    {
+        mLog->info ("onCancelRejected msg...");
+        mLog->info ("%s", msg.toString ().c_str ());
+    }
+
+    virtual void onMsg (uint64_t seqno, const cdr& msg)
+    {
+        mLog->info ("onMsg msg...");
+        mLog->info ("%s", msg.toString ().c_str ());
+    }
+
+    gwcConnector* mGwc;
+    logger* mLog;
+};
+
+int main (int argc, char** argv)
+{
+    properties p;
+    p.setProperty ("lh.console.level", "debug");
+    p.setProperty ("lh.console.color", "true");
+
+    std::string errorMessage;
+    bool ok = logService::get ().configure (p, errorMessage);
+    if (not ok)
+    {
+        std::string e("failed to configure logger: " + errorMessage);
+        errx (1, "%s", e.c_str ());
+    }
+
+    properties props (p, "gwc", "millennium", "sim");
+    props.setProperty ("host", "127.0.0.1:9899");
+    props.setProperty ("begin_string", "FIXT.1.1");
+    props.setProperty ("sender_comp_id", "DBL");
+    props.setProperty ("target_comp_id", "BME");
+    props.setProperty ("data_dictionary", "/home/colinp/dev/blu-corner/fosdk/examples/FIXT11.xml");
```

Comment:
Could this be a relative path?

Reply:
Aye - I'll switch to using the util function for getting it from an environment variable path